dashboard/model_layer.py

```diff
@@ -103,25 +126,51 @@ def get_available_nodes(limit=None):
     return nodes
 
 
-def get_node_status(node):
-    return 'Online' if node.is_active() else 'Offline'
+def get_node_hours_online(node_key, service_type, date_from, date_to) -> int:
+    query = models.NodeAvailability.query.with_entities(func.count()).filter(
+        models.NodeAvailability.node_key == node_key,
+        models.NodeAvailability.service_type == service_type,
+        date_from <= models.NodeAvailability.date,
+        models.NodeAvailability.date < date_to
+    )
+    records_count = query.scalar()
+    return int(round(records_count / 60.0))
 
 
-def get_node_info(node_key, service_type):
-    def get_node_time_online(day):
-        records_count = models.NodeAvailability.query.filter(
-            models.NodeAvailability.node_key == node_key,
-            models.NodeAvailability.service_type == service_type,
-            day <= models.NodeAvailability.date,
-            models.NodeAvailability.date < day+timedelta(days=1)
-        ).count()
+def get_registered_nodes(date_from, date_to):
+    query = models.Node.query.join(
+        models.IdentityRegistration,
+        models.IdentityRegistration.identity == models.Node.node_key,
+    ).filter(
+        date_from <= models.Node.updated_at,
+        models.Node.updated_at < date_to
```

Comment:
So if I was running a node from Monday to today (Friday), get_registered_nodes(Monday, Thursday) will not return me, because my node.updated_at == Friday, which is after `date_to`. Is this expected behaviour? :O

Reply:
`get_registered_nodes`  goes in pair with `date_from, date_to = get_week_range(datetime.utcnow().date())`
In your case `get_week_range(datetime.utcnow().date())` returns `previous Monday, next Monday` range